Runigram.java

```diff
@@ -133,9 +187,20 @@ public static Color[][] blend(Color[][] image1, Color[][] image2, double alpha)
 	 * of the source image.
 	 */
 	public static void morph(Color[][] source, Color[][] target, int n) {
-		//// Replace this comment with your code
+		if ((source.length != target.length) || (source[0].length != target[0].length)) {
+			target = scaled(target, source[0].length, source.length);
+		}
+		int index = 0;
+		while (index < n) {
+			double alpha = (double)((n - index) / (double)n);

```

Comment:
In Java if any of numerator or denominator is double the result is double - so the casting is not needed. So (n - index) / (double)n is enough